game/src/main/scala/io/aigar/game/Cell.scala

```diff
@@ -101,8 +108,20 @@ class Cell(val id: Int, player: Player, var position: Vector2 = new Vector2(0f,
     position.distanceTo(pos) <= radius
   }
 
-  def performAction(action: Action): Unit = {
+  def performAction(action: Action): Option[ScoreModification] = {
     target = action.target.toVector
+
+    val massToTrade = min(action.trade, max(mass - Cell.MinMass, 0)).toInt
```

Comment:
If possible, I think it would be nice if this logic would be in `trade` (or `tradeMass`, whichever you prefer):
```scala
val modifications = trade(action.trade)
```
Especially since we'll treat all the actions here, it would be nice to keep the action logic in their own methods (+ we reduce the chance that we'll call `tradeMass` without checking the mass first!)